gapis/perfetto/cc/processor.cpp

```diff
@@ -44,17 +41,105 @@ bool parse_data(processor processor, const void* data, size_t size) {
+      auto value = it.Get(col);
+
+      switch (desc->type() << 8 | value.type) {
+        // Nulls.
+        case p::QueryResult::ColumnDesc::UNKNOWN << 8 | ptp::SqlValue::kNull:
+          // Don't yet know the column type. Add a null value for each.
+          column->add_long_values(0);
+          column->add_double_values(0);
+          column->add_string_values("");
+          column->add_is_nulls(true);
+          break;
+        case p::QueryResult::ColumnDesc::LONG << 8 | ptp::SqlValue::kNull:
+          column->add_long_values(0);
+          column->add_is_nulls(true);
+          break;
+        case p::QueryResult::ColumnDesc::DOUBLE << 8 | ptp::SqlValue::kNull:
+          column->add_double_values(0);
+          column->add_is_nulls(true);
+          break;
+        case p::QueryResult::ColumnDesc::STRING << 8 | ptp::SqlValue::kNull:
+          column->add_string_values("");
+          column->add_is_nulls(true);
+          break;
+
+        // Values matching the type.
+        case p::QueryResult::ColumnDesc::UNKNOWN << 8 | ptp::SqlValue::kString:
+          desc->set_type(p::QueryResult::ColumnDesc::STRING);
+          column->clear_long_values();
+          column->clear_double_values();
+          // fall-through.
+        case p::QueryResult::ColumnDesc::STRING << 8 | ptp::SqlValue::kString:
+          column->add_string_values(value.string_value);
+          column->add_is_nulls(false);
+          break;
+        case p::QueryResult::ColumnDesc::UNKNOWN << 8 | ptp::SqlValue::kLong:
+          desc->set_type(p::QueryResult::ColumnDesc::LONG);
+          column->clear_string_values();
+          column->clear_double_values();
+          // fall-through.
+        case p::QueryResult::ColumnDesc::LONG << 8 | ptp::SqlValue::kLong:
+          column->add_long_values(value.long_value);
+          column->add_is_nulls(false);
+          break;
+        case p::QueryResult::ColumnDesc::UNKNOWN << 8 | ptp::SqlValue::kDouble:
+          desc->set_type(p::QueryResult::ColumnDesc::DOUBLE);
+          column->clear_string_values();
+          column->clear_long_values();
+          // fall-through.
+        case p::QueryResult::ColumnDesc::DOUBLE << 8 | ptp::SqlValue::kDouble:
+          column->add_double_values(value.double_value);
+          column->add_is_nulls(false);
+          break;
+
+        // Values needing conversion.
+        case p::QueryResult::ColumnDesc::LONG << 8 | ptp::SqlValue::kDouble:
+          // TODO: should we "upgrade" the column to double?
+          column->add_long_values(static_cast<int64_t>(value.double_value));
+          column->add_is_nulls(false);
+          break;
+        case p::QueryResult::ColumnDesc::DOUBLE << 8 | ptp::SqlValue::kLong:
+          column->add_double_values(static_cast<double>(value.long_value));
+          column->add_is_nulls(false);
+          break;
+        default:
+          // ignore mismatched numeric/string values.
```

Comment:
I'd rather not add a print statement here, because:
1) we're in cgo land here and going to go just to log something seems expensive
2) this is in a tight loop reading rows from a query result and could cause a lot of messages very quickly.